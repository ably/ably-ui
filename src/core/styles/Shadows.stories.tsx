import React from "react";

export default {
  title: "Styles/Shadows",
};

const shadowClasses = {
  soft: [
    "ui-shadow-xs-soft",
    "ui-shadow-sm-soft",
    "ui-shadow-md-soft",
    "ui-shadow-lg-soft",
    "ui-shadow-xl-soft",
  ],
  medium: [
    "ui-shadow-xs-medium",
    "ui-shadow-sm-medium",
    "ui-shadow-md-medium",
    "ui-shadow-lg-medium",
    "ui-shadow-xl-medium",
  ],
  strong: [
    "ui-shadow-xs-strong",
    "ui-shadow-sm-strong",
    "ui-shadow-md-strong",
    "ui-shadow-lg-strong",
    "ui-shadow-xl-strong",
  ],
};

const shadowRow = (weight: "soft" | "medium" | "strong") => (
  <div className="flex gap-8">
    {shadowClasses[weight].map((shadowClass) => (
      <div
        key={shadowClass}
        className={`${weight === "strong" ? "bg-neutral-1100 text-white" : "bg-white"} rounded-lg w-16 h-16 ${shadowClass} flex items-center justify-center ui-text-p1`}
      >
        {shadowClass.split("-")[2]}
      </div>
    ))}
  </div>
);

export const Soft = {
  render: () => shadowRow("soft"),
  parameters: {
    docs: {
      description: {
        story: "Example usage: `.ui-shadow-xs-soft`",
      },
    },
  },
};

export const Medium = {
  render: () => shadowRow("medium"),
  parameters: {
    docs: {
      description: {
        story: "Example usage: `.ui-shadow-xs-medium`",
      },
    },
  },
};

export const Strong = {
  render: () => shadowRow("strong"),
  parameters: {
    docs: {
      description: {
        story: "Example usage: `.ui-shadow-xs-strong`",
      },
    },
  },
};
