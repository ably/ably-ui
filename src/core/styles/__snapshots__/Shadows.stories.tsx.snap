// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles/Shadows Medium smoke-test 1`] = `
<div class="flex gap-32">
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-xs-medium flex items-center justify-center ui-text-p1">
    xs
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-sm-medium flex items-center justify-center ui-text-p1">
    sm
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-md-medium flex items-center justify-center ui-text-p1">
    md
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-lg-medium flex items-center justify-center ui-text-p1">
    lg
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-xl-medium flex items-center justify-center ui-text-p1">
    xl
  </div>
</div>
`;

exports[`Styles/Shadows Soft smoke-test 1`] = `
<div class="flex gap-32">
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-xs-soft flex items-center justify-center ui-text-p1">
    xs
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-sm-soft flex items-center justify-center ui-text-p1">
    sm
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-md-soft flex items-center justify-center ui-text-p1">
    md
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-lg-soft flex items-center justify-center ui-text-p1">
    lg
  </div>
  <div class="bg-white rounded-lg w-64 h-64 ui-shadow-xl-soft flex items-center justify-center ui-text-p1">
    xl
  </div>
</div>
`;

exports[`Styles/Shadows Strong smoke-test 1`] = `
<div class="flex gap-32">
  <div class="bg-neutral-1100 text-white rounded-lg w-64 h-64 ui-shadow-xs-strong flex items-center justify-center ui-text-p1">
    xs
  </div>
  <div class="bg-neutral-1100 text-white rounded-lg w-64 h-64 ui-shadow-sm-strong flex items-center justify-center ui-text-p1">
    sm
  </div>
  <div class="bg-neutral-1100 text-white rounded-lg w-64 h-64 ui-shadow-md-strong flex items-center justify-center ui-text-p1">
    md
  </div>
  <div class="bg-neutral-1100 text-white rounded-lg w-64 h-64 ui-shadow-lg-strong flex items-center justify-center ui-text-p1">
    lg
  </div>
  <div class="bg-neutral-1100 text-white rounded-lg w-64 h-64 ui-shadow-xl-strong flex items-center justify-center ui-text-p1">
    xl
  </div>
</div>
`;
